app/src/main/java/org/schabi/newpipe/App.java

```diff
@@ -117,6 +119,7 @@ public void onTerminate() {
             disposable.dispose();
         }
         super.onTerminate();
+        PicassoHelper.terminate();
```

Comment:
Just wondering why are you are calling PicassoHelper.terminate() after super.onTerminate() call and not before

Reply:
I don't know if it makes any difference, but I think I chose to do it this way because there could still be something doing image processing before `super.onTerminate()`, but surely there isn't after it.

Reply:
I see it seems it is totally irrelevant what is done in this method on a real device. It is never called according to:
https://developer.android.com/reference/android/app/Application#onTerminate()

Reply:
Mmmh... should we remove it? @TobiGr

Reply:
I think so. That's at least how I understand the docs